### ISSUE TITLE
Cannot create a stable cluster

### ISSUE THREAD

**Issue by ariselseng** [colinmollenhour/mariadb-galera-swarm#33]
I am using docker swarm. I did exactly as in the docker swarm example, except removing remoing the NODE_ADDRESS variable to let it autoconfigure.
I also feel like i have tried everything. Including making the network first and inspecting it to get the ip range, and using that for NODE_ADDRESS variable.
Does this project work in its current state?
```shell
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592977786624 [Note] WSREP: save pc into disk
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592969393920 [Note] WSREP: New COMPONENT: primary = yes, bootstrap = no, my_idx = 1, memb_num = 2
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592969393920 [Note] WSREP: STATE EXCHANGE: Waiting for state UUID.
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592977786624 [Note] WSREP: forgetting b663bab1 (tcp://10.0.4.17:4567)
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592969393920 [Note] WSREP: STATE EXCHANGE: sent state msg: f3f2f61c-1ba1-11e8-a872-5f4a2014eb6d
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592969393920 [Note] WSREP: STATE EXCHANGE: got state msg: f3f2f61c-1ba1-11e8-a872-5f4a2014eb6d from 0 (5dcd2316d4f6)
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592969393920 [Note] WSREP: STATE EXCHANGE: got state msg: f3f2f61c-1ba1-11e8-a872-5f4a2014eb6d from 1 (f6c9099d7e52)
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592969393920 [Note] WSREP: Quorum results:
galera_node.2.hfps0ofumgos@swarm-master-03    |         version    = 4,
galera_node.2.hfps0ofumgos@swarm-master-03    |         component  = PRIMARY,
galera_node.2.hfps0ofumgos@swarm-master-03    |         conf_id    = 11,
galera_node.2.hfps0ofumgos@swarm-master-03    |         members    = 1/2 (joined/total),
galera_node.2.hfps0ofumgos@swarm-master-03    |         act_id     = 0,
galera_node.2.hfps0ofumgos@swarm-master-03    |         last_appl. = 0,
galera_node.2.hfps0ofumgos@swarm-master-03    |         protocols  = 0/8/3 (gcs/repl/appl),
galera_node.2.hfps0ofumgos@swarm-master-03    |         group UUID = a156782c-1ba0-11e8-9506-f3d6550b8b55
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592969393920 [Note] WSREP: Flow-control interval: [23, 23]
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:23 140592969393920 [Note] WSREP: Trying to continue unpaused monitor
   joiner: => Rate:[   0 B/s] Avg:[   0 B/s] Elapsed:0:01:40  Bytes:    0 B ] Avg:[   0 B/s] Elapsed:0:00:10  Bytes:    0 B 
galera_node.2.hfps0ofumgos@swarm-master-03    | WSREP_SST: [ERROR] Possible timeout in receving first data from donor in gtid stage (20180227 09:38:28.182)
galera_node.2.hfps0ofumgos@swarm-master-03    | WSREP_SST: [ERROR] Cleanup after exit with status:32 (20180227 09:38:28.186)
galera_node.2.hfps0ofumgos@swarm-master-03    | WSREP_SST: [INFO] Cleaning up fifo file /tmp/mysql-console/fifo (20180227 09:38:28.191)
galera_node.2.hfps0ofumgos@swarm-master-03    | rm: cannot remove '/tmp/mysql-console/fifo': Permission denied
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:28 140592961001216 [ERROR] WSREP: Failed to read 'ready <addr>' from: wsrep_sst_xtrabackup-v2 --role 'joiner' --address '10.0.4.16' --datadir '/var/lib/mysql/'   --parent '41'  '' 
galera_node.2.hfps0ofumgos@swarm-master-03    |         Read: '(null)'
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:28 140592961001216 [ERROR] WSREP: Process completed with error: wsrep_sst_xtrabackup-v2 --role 'joiner' --address '10.0.4.16' --datadir '/var/lib/mysql/'   --parent '41'  '' : 1 (Operation not permitted)
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:28 140595377510144 [ERROR] WSREP: Failed to prepare for 'xtrabackup-v2' SST. Unrecoverable.
galera_node.2.hfps0ofumgos@swarm-master-03    | 2018-02-27  9:38:28 140595377510144 [ERROR] Aborting
galera_node.2.hfps0ofumgos@swarm-master-03    | 
galera_node.1.h1m8gi66g9va@node-02    | Error in my_thread_global_end(): 1 threads didn't exit
galera_node.1.h1m8gi66g9va@node-02    | MariaDB exited with return code (0)
galera_node.1.h1m8gi66g9va@node-02    | # GALERA saved state
galera_node.1.h1m8gi66g9va@node-02    | version: 2.1
galera_node.1.h1m8gi66g9va@node-02    | uuid:    00000000-0000-0000-0000-000000000000
galera_node.1.h1m8gi66g9va@node-02    | seqno:   -1
galera_node.1.h1m8gi66g9va@node-02    | safe_to_bootstrap: 1
galera_node.1.h1m8gi66g9va@node-02    | Goodbye

**Comment by ariselseng** [comment]
I forgot to mention that the nodes only live for 2 minutes. After that new containers replaces them.

**Comment by smidge84** [comment]
I am having exactly the same issue. I also noticed that once this has occurred, then the seed node starts failing it's health check. Portainer reports that `curl: (22) The requested URL returned error: 503 Service Unavailable` which results in the seed container being cycled. I'm trying to read more about galera and xtrabackup in general to try and understand this part of the process and why the comms hang.
Some insight would very much be appreciated.

**Comment by peter-slovak** [comment]
@cowai @smidge84 You're probably hitting https://jira.mariadb.org/browse/MDEV-15383?workflowName=MariaDB+v3&stepId=1 and https://jira.mariadb.org/browse/MDEV-15254 . It seems there are more patches coming, similar to the 10.1.31 release. My personal advice is using the 10.1.31 Dockerfile - I've already spent a few hours trying to get 10.2 running, without success.

**Comment by ariselseng** [comment]
@peter-slovak 
Thank you for the tip.
Just to confirm. Would the tag "10.1.31-2018-02-20" work okay?

**Comment by colinmollenhour** [comment]
Yes, I'm fairly confident that 10.1.31-2018-02-20 is stable as it applies some patches to fix the major regressions:

https://github.com/colinmollenhour/mariadb-galera-swarm/blob/master/Dockerfile-10.1#L22

I was able to bootstrap my Kontena cluster using this patched 10.1.31 version. I've been told 10.2 works fine but I have not tried it myself.

**Comment by ariselseng** [comment]
@colinmollenhour I tried that image, and it has been stable for days now. Thanks! You may want to remove the 10.2 image or warn against it not working. Just a thought.

**Comment by FalkNisius** [comment]
In my case I found out that in the 10.2 Mariadb image, the tail command used to watch /tmp/mysql../fifo fails, if the swarm node use the overlayfs. The tail from coreutils (8.23.4) in this image has a known bug, that this file system type is unknown. A newer debian brings coreutils 8.24.1, where that is fixed. Perhaps someone can confirm this in a test against aufs as overlay fs.

**Comment by ariselseng** [comment]
I use aufs and had other problems so this is probably not related. Thanks for the heads up as I wanted to go overlayfs very soon.

**Comment by gentunian** [comment]
> Yes, I'm fairly confident that 10.1.31-2018-02-20 is stable as it applies some patches to fix the major regressions:
> 
> https://github.com/colinmollenhour/mariadb-galera-swarm/blob/master/Dockerfile-10.1#L22
> 
> I was able to bootstrap my Kontena cluster using this patched 10.1.31 version. I've been told 10.2 works fine but I have not tried it myself.

It seems that now that image is not working. Setting `colinmollenhour/mariadb-galera-swarm:10.1.31-2018-02-20` in the `docker-compose.yml` file will abort with this error:

```
InnoDB: No valid checkpoint found.
InnoDB: A downgrade from MariaDB 10.2.2 or later is not supported.
InnoDB: If this error appears when you are creating an InnoDB database,
InnoDB: the problem may be that during an earlier attempt you managed
InnoDB: to create the InnoDB data files, but log file creation failed.
InnoDB: If that is the case, please refer to
InnoDB: http://dev.mysql.com/doc/refman/5.6/en/error-creating-innodb.html
2019-04-18 16:01:46 140073899898816 [ERROR] Plugin 'InnoDB' init function returned error.
2019-04-18 16:01:46 140073899898816 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2019-04-18 16:01:46 140073899898816 [Note] Plugin 'FEEDBACK' is disabled.
2019-04-18 16:01:46 140073899898816 [ERROR] Unknown/unsupported storage engine: innodb
2019-04-18 16:01:46 140073899898816 [ERROR] Aborting

Error in my_thread_global_end(): 1 threads didn't exit
MariaDB exited with return code (0)
# GALERA saved state
version: 2.1
uuid:    7d806cd0-61ee-11e9-b306-43433cf9bcf9
seqno:   -1
safe_to_bootstrap: 
```

I'm having a hard time to make the example work, so I found this issue and tried this image. Seems that this doesn't work either.
